packages/react/src/components/ContainedList/ContainedList.tsx

```diff
@@ -5,14 +5,51 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { ReactNode } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { LayoutConstraint } from '../Layout';
 import { useId } from '../../internal/useId';
 import { usePrefix } from '../../internal/usePrefix';
 
-const variants = ['on-page', 'disclosed'];
+const variants: ('on-page' | 'disclosed')[] = ['on-page', 'disclosed'];
+
+interface ContainedListProps {
+  /**
+   * A slot for a possible interactive element to render.
+   */
+  action?: ReactNode;
+
+  /**
+   * A collection of ContainedListItems to be rendered in the ContainedList
+   */
+  children?: ReactNode;
+
+  /**
+   * Additional CSS class names.
+   */
+  className?: string;
+
+  /**
+   * Specify whether the dividing lines in between list items should be inset.
+   */
+  isInset?: boolean;
+
+  /**
+   * The kind of ContainedList you want to display
+   */
+  kind?: 'on-page' | 'disclosed';
+
+  /**
+   * A label describing the contained list.
+   */
+  label: ReactNode;
```

Comment:
This accepts a string too, right?

```suggestion
  label: string | ReactNode;
```

Reply:
Yes, I overlooked it, will make the changes.